plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -85,9 +86,7 @@ public PartitionTable(SchemaTableName tableName, TypeManager typeManager, Table
         this.typeManager = requireNonNull(typeManager, "typeManager is null");
         this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
         this.snapshotId = requireNonNull(snapshotId, "snapshotId is null");
-        this.idToTypeMapping = icebergTable.schema().columns().stream()
-                .filter(column -> column.type().isPrimitiveType())
-                .collect(Collectors.toMap(Types.NestedField::fieldId, (column) -> column.type().asPrimitiveType()));
+        this.idToTypeMapping = primitiveFieldTypes(icebergTable.schema());
```

Comment:
We should somehow indicate/note that the `idToTypeMapping` contains the source fields instead of transformed ones.

Reply:
> We should somehow indicate/note that the idToTypeMapping contains the source fields instead of transformed ones.

this didn't change, right?

> Only PartitionField refers to transformed field and to get to source field you need to use PartitionField#sourceId.

We could narrow the map to only partition fields (and rename appropriately) , is that what you mean here?

Reply:
Nothing changed - I'm just expressing frustration at something which I had to spend a bit of time fighting when debugging a past issue. 🙂 

I don't think we can do anything other than people being aware of the difference between `PartitionField` and `NestedField` even though they sound the same.